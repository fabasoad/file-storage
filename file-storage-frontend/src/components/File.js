import React from 'react';

const File = (props) => (
    <div className="file">
        <p className="file__text">{props.count}. {props.fileName}</p>
        <button
            className="button button--link"
            onClick={(e) => {
                props.handleDeleteFile(props.fileName);
            }}
        >
            Remove
        </button>
    </div>
);

export default File;