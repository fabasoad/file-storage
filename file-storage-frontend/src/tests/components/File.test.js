import React from 'react';
import { shallow } from 'enzyme';
import File from '../../components/File';

let wrapper, expectedFileName, expectedCount, handleDeleteFile;

beforeEach(() => {
    expectedFileName = 'testFileName';
    expectedCount = 12;
    handleDeleteFile = jest.fn();
    wrapper = shallow(<File
        key={expectedFileName}
        fileName={expectedFileName}
        count={expectedCount}
        handleDeleteFile={handleDeleteFile}
    />);
});

test('should render File correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle handleDeleteFile', () => {
    wrapper.find('button').prop('onClick')();
    expect(handleDeleteFile).toHaveBeenLastCalledWith(expectedFileName);
});