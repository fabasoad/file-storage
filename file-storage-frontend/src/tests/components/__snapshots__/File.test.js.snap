// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render File correctly 1`] = `
<div
  className="file"
>
  <p
    className="file__text"
  >
    12
    . 
    testFileName
  </p>
  <button
    className="button button--link"
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
